### PR TITLE
Add dependency on ZIO and remove effect module

### PR DESCRIPTION
**Pull request by ktonga**

### CODE REVIEW

File: base/shared/src/main/scala/scalaz/core/io.scala [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+import java.lang.Throwable
+
+import com.github.ghik.silencer.silent
+
+import algebra.MonoidClass
+import ct._
+import types.IsCovariantClass
+import zio.{ IO, Fiber }
+
+trait IOModule
```
Comment (hrhino) [inline base/shared/src/main/scala/scalaz/core/io.scala]: needed?

File: base/shared/src/main/scala/scalaz/core/io.scala [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+      final def map[A, B](ma: Fiber[E, A])(f: A => B): Fiber[E, B] = new Fiber[E, B] {
+        def join: IO[E, B]                            = ma.join.map(f)
+        def interrupt[E2](t: Throwable): IO[E2, Unit] = ma.interrupt(t)
+      }
+    })
+}
+
+@silent
+private[core] object IOImpl extends IOModule with IOInstances with FiberInstances
```
Comment (hrhino) [inline base/shared/src/main/scala/scalaz/core/io.scala]: likewise, what does this do?

File: base/shared/src/main/scala/scalaz/core/io.scala (hunk 1 above)
Comment (ktonga) [reply base/shared/src/main/scala/scalaz/core/io.scala]: Don't think so, just following conventions. Can remove it.

**Comment by ktonga** [comment]
Ping @jdegoes

**Comment by jdegoes** [comment]
@ktonga Nicely done! Any ideas on how to keep this up to date with the latest ZIO?

**Comment by ktonga** [comment]
Actually I'm more concerned about the opposite, it will be kept *too* up to date, since we depend on a SNAPSHOT and it will be re-checked on every scalaz build.

To make it a bit more deterministic i'd suggest:

1. Ideally, we introduce source dependency (as @fommil has been suggesting ad nauseum :smile: ) and we move the head of the submodule as required. Dunno how mature support  for that in scala is right now so maybe something to check again in the future.
2. A more realistic solution would be to start using semantic versioning in ZIO and use a version expression in scalaz instead of a fixed one.

**Pull request merged**